Source/Core/Core/WiiRoot.cpp

```diff
@@ -312,29 +336,45 @@ void InitializeWiiFileSystemContents(
   if (!CopySysmenuFilesToFS(fs.get(), File::GetSysDirectory() + WII_USER_DIR, ""))
     WARN_LOG_FMT(CORE, "Failed to copy initial System Menu files to the NAND");
 
-  if (WiiRootIsTemporary())
+  const bool is_temp_nand = WiiRootIsTemporary();
+  if (is_temp_nand)
   {
     // Generate a SYSCONF with default settings for the temporary Wii NAND.
     SysConf sysconf{fs};
     sysconf.Save();
 
     InitializeDeterministicWiiSaves(fs.get(), boot_session_data);
   }
-  else if (save_redirect)
+
+  if (save_redirect)
   {
     const u64 title_id = SConfig::GetInstance().GetTitleID();
     std::string source_path = Common::GetTitleDataPath(title_id);
-    if (!File::IsDirectory(save_redirect->m_target_path))
+
+    DiscIO::Riivolution::SavegameRedirect evaluated_redirect;
```

Comment:
Why not modify `save_redirect` directly? You could avoid to need for a separate DiscIO::Riivolution::SavegameRedirect variable, a separate move and it's more error prone to have both `save_redirect` and `evaluated_redirect` in the same scope. Too easy to accidentally use the wrong one...

If you use `save_redirect`, you'd only have to overwrite the SavegameRedirect's string in the `if (s_temp_nand)` instead of having to create an entirely new SavegameRedirect and initialize each field (even fields which haven't changed, like `m_clone`) manually.

Reply:
Yeah, that's fair.